back/config/environments/test.rb

```diff
@@ -12,7 +12,7 @@
   # Do not eager load code on boot. This avoids loading your whole application
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
-  config.eager_load = true
+  config.eager_load = ENV['CI'].present?
```

Comment:
Why do they say "If you are using a tool that preloads Rails for running tests, you may have to set it to true"?

Maybe it won't work if you:
1. Run a test (with Spring)
2. Add a new model class
3. Run a test with that model class
4. (perhaps) The new model will not be loaded.

Can it be the case? Here they mention this trick https://stackoverflow.com/a/54226404

Reply:
@alexander-cit I tried your scenario and indeed it produces a NameError. I prefer reverting this change instead of adding a dependency on the `ENABLE_SPRING` env var, because of 2 reasons:
* `config.eager_load = !!(ENV['ENABLE_SPRING'] == 'true')` would have to be combined with `ENV['CI'].present?`, because `ENABLE_SPRING` is not set on CI. The condition is becoming too complex for my taste.
* The biggest win comes from using Spring anyway.

wdyt? And what does @nTraum think?

Reply:
In my opinion, `ENV['CI'].present? || ENV['ENABLE_SPRING'] == 'true'` doesn't look too crazy. But I agree that half a second doesn't matter that much when a test runs for 18 seconds (without spring).

Reply:
I'm slightly leaning to the `CI` env var solution as it is implemented right now. Have seen this in many test setups & CI services already, I think it's a common solution.

Reply:
I reverted the change.

Reply:
@alexander-cit About your scenario. Does that work for you today? When I follow the scenario using the master, the new model is not loaded. That is not a surprise. [The documentation states](https://github.com/rails/spring#enable-reloading) that In a Spring context `config.cache_classes = false` should be part of the configuration, but it is `config.cache_classes = true` in `back/config/environments/test.rb`.

I changed it to `false` and now the new model is loaded. Conclusion: that stackoverflow article is wrong.

If you and @nTraum are fine with this change, I will apply it.